src/components/application_manager/src/message_helper/message_helper.cc

```diff
@@ -638,8 +638,11 @@ smart_objects::SmartObjectSPtr MessageHelper::CreateDeviceListSO(
   smart_objects::SmartObject& list_so = (*device_list_so)[strings::device_list];
 
   int32_t index = 0;
+  // According to requirements, SDL should send info about 100 devices at
+  // maximum, even if SDL has more devices connected.
+  const int32_t max_device_count = 100;
```

Comment:
@LitvinenkoIra I suppose device count can not be negative number as well as index